pymeasure/instruments/agilent/agilentB1500.py

```diff
@@ -1590,6 +1655,238 @@ def __init__(self, smu_type):
         self.meas = Ranging(supported_meas_ranges, ranges, fixed_ranges=True)
 
 
+######################################
+# SPGU Setup
+######################################
+
+
+class SPGU:
```

Comment:
This is a channel, right, at least in an abstract way? So please inherit from the `Channel` class!

See: https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments/channels.html

There is no problem to have Channels of Channels. They were designed that way.

You might even add these channels to the api documentation.